### ISSUE TITLE
path internal handling is weird

### ISSUE THREAD

**Issue by ofZach** [openframeworks/projectGenerator#6]
there's some stuff in ofAddon (it's in the main repo, but I'm putting the issue here so it's easier to track) that seems kind of wrong to me.  Paths are coming in either unix-y or windows-y, but it feels like we should make the decision to convert all paths to unix internally and only convert back when and if we need to at the project level (ie, visualStudioProject should make sure it outputs windows paths)

I'm talking specifically about stuff like this: 

https://github.com/openframeworks/openFrameworks/blob/master/addons/ofxProjectGenerator/src/addons/ofAddon.cpp#L428-L432

which is making some assumptions and trying to be a bit smart about paths....   My intuition is that things like generating an OSX project on windows could break in weird ways here....

the tl;dr is - let's make sure to just use unix under the hood and convert to windows only when absolutely necessary.

**Comment by arturoc** [comment]
we shouldn't even use one or the other just use ofFile, ofFileSystemPath or std::filesystem::path to work with paths which are mulitplatform and then print out with the correct separators when generating the projects
